### PR TITLE
Add builtin sorted

### PR DESCRIPTION
**Pull request by S-YOU**
Add builtin sorted, without cmp, key, and reverse options.

### CODE REVIEW

**Comment by S-YOU** [comment]
Thanks @meadori, I didn't realize there is tests for python side.
--
Looks like I shouldn't include reversed in this PR. I didnt know about there is __reverse__ slot.

**Comment by meadori** [comment]
Yeah, I would break the `reversed` one out into a separate PR.  It will probably be more involved.

**Comment by S-YOU** [comment]
@meadori, Thank you, removed reversed

File: runtime/builtin_types.go [hunk 1]
```diff
@@ -541,6 +541,18 @@ func builtinRepr(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return s.ToObject(), nil
 }
 
+func builtinSorted(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	if raised := checkFunctionArgs(f, "sorted", args, ObjectType); raised != nil {
+		return nil, raised
+	}
+	result, raised := listNew(f, ListType, Args{args[0]}, KWArgs{})
```
Comment (trotterdylan) [inline runtime/builtin_types.go]: `ListType.Call(f, Args{args[0]}, nil)`

File: runtime/builtin_types.go [hunk 2]
```diff
@@ -541,6 +541,18 @@ func builtinRepr(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return s.ToObject(), nil
 }
 
+func builtinSorted(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	if raised := checkFunctionArgs(f, "sorted", args, ObjectType); raised != nil {
```
Comment (trotterdylan) [inline runtime/builtin_types.go]: `sorted()` is a little more complicated than this:

```
sorted(iterable, cmp=None, key=None, reverse=False) --> new sorted list
```

A TODO is fine. Then that functionality could be added to list.sort() and then passed when you call list.sort() below.

File: runtime/builtin_types_test.go [hunk 3]
```diff
@@ -228,6 +228,13 @@ func TestBuiltinFuncs(t *testing.T) {
 		{f: "repr", args: wrapArgs(NewUnicode("abc")), want: NewStr("u'abc'").ToObject()},
 		{f: "repr", args: wrapArgs(newTestTuple("foo", "bar")), want: NewStr("('foo', 'bar')").ToObject()},
 		{f: "repr", args: wrapArgs("a", "b", "c"), wantExc: mustCreateException(TypeErrorType, "'repr' requires 1 arguments")},
+		{f: "sorted", args: wrapArgs(NewList()), want: NewList().ToObject()},
+		{f: "sorted", args: wrapArgs(newTestList("foo", "bar")), want: newTestList("bar", "foo").ToObject()},
```
Comment (trotterdylan) [inline runtime/builtin_types_test.go]: Maybe some test cases that are other iterable types like dict and tuple.

File: runtime/builtin_types.go (hunk 2 above)
Comment (meadori) [inline runtime/builtin_types.go]: I noticed that as well, but didn't explicitly mention it b/c @S-YOU put `Add builtin sorted, without cmp, key, and reverse options.` in the change request message.

**Comment by S-YOU** [comment]
Updated following
- TODO about (cmp=None, key=None, reverse=False)
- Use ListType.Call instead of listNew
- Added tests for Tuple and Dict
- Added some python side testing

Please take a look, @meadori, @trotterdylan

File: runtime/builtin_types.go [hunk 4]
```diff
@@ -541,6 +541,19 @@ func builtinRepr(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return s.ToObject(), nil
 }
 
+func builtinSorted(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	//TODO: Support (cmp=None, key=None, reverse=False)
```
Comment (trotterdylan) [inline runtime/builtin_types.go]: Space after `//` in comments:

`// TODO: ...`

**Comment by S-YOU** [comment]
@trotterdylan, added space after comment //, thank you very much.

**Pull request merged**